data-serving/data-service/test/case.test.ts

```diff
@@ -36,110 +33,77 @@ describe('GET', () => {
 
         expect(res.body._id).toEqual(c._id.toString());
     });
-    it('one absent item should return 404 NOT FOUND', (done) => {
-        request(app)
+    it('one absent item should return 404 NOT FOUND', () => {
+        return request(app)
             .get('/api/cases/53cb6b9b4f4ddef1ad47f943')
-            .expect(404, done);
+            .expect(404);
     });
-    it('list should return 200 OK', (done) => {
-        request(app)
+    it('list should return 200 OK', () => {
+        return request(app)
```

Comment:
Just curious, why switch to returning Promises in this file? I'm guessing this'll cause the test framework to wait for each case to resolve, whereas using callbacks won't (IIUC, this is why the callback style is more popular with supertest).

Reply:
I personally think it's cleaner and easier to trace the flow -- I generally strongly prefer promises to callbacks. From what I've read, it also makes error messages clearer.

But this might be just my prior test framework preference -- if this isn't a very supertest-friendly, I can change it.

Can you explain a little more about waiting for the cases to resolve? Don't we generally call done() at the some point at which we'd be returning the promise?

Reply:
Yeah, I generally prefer promises to callbacks as well. Looking back, the article I'm cribbing is for supertest with an older version of Mocha (whereas we're using (the current version of) Jest). There shouldn't be a material difference; this structure SGTM.